examples/core/transform-feedback-instanced/app.js

```diff
@@ -0,0 +1,406 @@
+
+    const modelRender = new Model(gl, {
+      id: 'Model-Render',
+      program: new Program(gl, {
+        vs: VS_DRAW,
+        fs: FS_DRAW
+      }),
+      geometry: new Geometry({
+        id: 'Model-Render-Geometry',
+        drawMode: gl.TRIANGLE_FAN,
+        vertexCount: 3
+      }),
+      isInstanced: true,
+      instanceCount: NUM_INSTANCES,
+      isIndexed: false
+    });
+
+    const modelTransform = new Model(gl, {
+      id: 'Model-Transform',
+      program: new Program(gl, {
+        vs: VS_EMIT,
+        fs: FS_EMIT,
+        varyings: VARYINGS
+      }),
+      geometry: new Geometry({
+        id: 'Model-Transform-Geometry',
+        drawMode: gl.POINTS,
+        vertexCount: NUM_INSTANCES
+      }),
+      isInstanced: false,
+      isIndexed: false
+    });
+
+    // -- Initialize data
+    const trianglePositions = new Float32Array([
+      0.015, 0.0,
+      -0.010, 0.010,
+      -0.010, -0.010
+    ]);
+
+    const instanceOffsets = new Float32Array(NUM_INSTANCES * 2);
+    const instanceRotations = new Float32Array(NUM_INSTANCES);
+    const instanceColors = new Float32Array(NUM_INSTANCES * 3);
+
+    for (let i = 0; i < NUM_INSTANCES; ++i) {
+      const oi = i * 2;
+      const ri = i;
+      const ci = i * 3;
+
+      instanceOffsets[oi] = Math.random() * 2.0 - 1.0;
+      instanceOffsets[oi + 1] = Math.random() * 2.0 - 1.0;
+
+      instanceRotations[ri] = Math.random() * 2 * Math.PI;
+
+      instanceColors[ci] = Math.random();
+      instanceColors[ci + 1] = Math.random();
+      instanceColors[ci + 2] = Math.random();
+    }
+
+    // -- Init VertexArrays and TransformFeedback objects
+
+    const vertexArrays = new Array(2);
+    const transformVertexArrays = new Array(2);
+    const transformFeedbacks = new Array(2);
+
+    const positionBuffer = new Buffer(gl, {
+      data: trianglePositions,
+      size: 2,
+      type: gl.FLOAT,
+      normalized: false,
+      stride: 0,
+      offset: 0,
+      instanced: 0,
+      usage: gl.STATIC_DRAW
+    });
+
+    const colorBuffer = new Buffer(gl, {
+      data: instanceColors,
+      size: 3,
+      type: gl.FLOAT,
+      normalized: false,
+      stride: 0,
+      offset: 0,
+      instanced: 1,
+      usage: gl.STATIC_DRAW
+    });
+
+    for (let va = 0; va < vertexArrays.length; ++va) {
+      const offsetBuffer = new Buffer(gl, {
+        data: instanceOffsets,
+        size: 2,
+        type: gl.FLOAT,
+        normalized: false,
+        stride: 0,
+        offset: 0,
+        instanced: 1,
+        usage: gl.STREAM_COPY
+      });
+
+      const rotationBuffer = new Buffer(gl, {
+        data: instanceRotations,
+        size: 1,
+        type: gl.FLOAT,
+        normalized: false,
+        stride: 0,
+        offset: 0,
+        instanced: 1,
+        usage: gl.STREAM_COPY
+      });
+
+      vertexArrays[va] = new VertexArray(gl, {
+        buffers: {
+          [COLOR_LOCATION]: {
+            buffer: colorBuffer,
+            size: 3,
```

Comment:
The intention is that these parameters can be read from the `Buffer` object here at bind time. 

I believe that this is the only reason we store those params in the `Buffer` object, since WebGLBuffer itself doesn't do anything with them.

At least we should not specify them twice. Either delete them here, or in buffer creation.

Reply:
Removed all default settings except `instanced`,  this flag is switched on some buffers between transform and regular render cycles, setting it always to a value for clarity.